safekeeper/src/wal_backup_partial.rs

```diff
@@ -270,8 +266,26 @@ impl PartialBackup {
     }
 }
 
+/// Check if everything is uploaded and partial backup task doesn't need to run.
+pub fn needs_uploading(state: &StateSnapshot, uploaded: &Option<PartialRemoteSegment>) -> bool {
+    match uploaded {
+        Some(uploaded) => {
+            uploaded.status != UploadStatus::Uploaded
+                || uploaded.flush_lsn != state.flush_lsn
+                || uploaded.commit_lsn != state.commit_lsn
+                || uploaded.term != state.term
+        }
+        None => true,
+    }
+}
+
+/// Main task for partial backup. It waits for the flush_lsn to change and then uploads the
```

Comment:
Hmm, didn't think about it. Do you have ideas for a new name in mind?

Actually, we have "WAL backup" and "partial WAL backup" tasks. Both tasks are uploading files to remote storage before deleting it from disk, and it was always the case with "WAL backup". It means technically "WAL backup" already works in a way that remote storage is the only copy of the data rather than a backup.

Reply:
Opened an issue about this: https://github.com/neondatabase/neon/issues/8146